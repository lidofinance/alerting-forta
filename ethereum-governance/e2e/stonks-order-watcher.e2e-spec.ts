import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 180_000; // ms

/**
 * Tests works for testflight stonks 0x5FA801ee2202b3Bcd2317F9a65A408A725746647
 */

describe.skip("treasury-swap e2e tests", () => {
  let runBlock: (
    blockHashOrNumber: string | number,
    initBlock?: number,
    skipInit?: boolean,
  ) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;
  let timeSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
    timeSpy = jest.spyOn(Date, "now");
    timeSpy.mockImplementation(() => new Date("2023-12-31"));
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(
        provideAgentPath("subagents/stonks-order-watcher/stonks-order-watcher"),
      ),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should find 0 testflight stonks at creation block",
    async () => {
      let findings = await runBlock(19440190, 19440050);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );
  it(
    "should find 1 testflight stonks at creation block",
    async () => {
      let findings = await runBlock(19440200, 19440050);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should find 0 testflight stonks at next after creation block",
    async () => {
      let findings = await runBlock(19440190, 19440070);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should find 1 testflight stonks at next after creation block",
    async () => {
      let findings = await runBlock(19440200, 19440070);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should find 0 stonks in 30+ after testflight",
    async () => {
      let findings = await runBlock(19440220, 19440210);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should find 0 in 31+ after testflight",
    async () => {
      let findings = await runBlock(19440680, 19440210);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );

  it(
    "should find 0 in 120+ after testflight",
    async () => {
      let findings = await runBlock(19440680, 19440660);
      findings.sort((a, b) => (a.description < b.description ? -1 : 1));
      expect(findings).toMatchSnapshot();
    },
    TEST_TIMEOUT,
  );
});
