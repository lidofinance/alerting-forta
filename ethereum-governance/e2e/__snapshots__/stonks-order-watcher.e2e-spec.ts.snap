// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`treasury-swap e2e tests should find 0 in 31+ after testflight 1`] = `[]`;

exports[`treasury-swap e2e tests should find 0 in 120+ after testflight 1`] = `[]`;

exports[`treasury-swap e2e tests should find 0 stonks in 30+ after testflight 1`] = `[]`;

exports[`treasury-swap e2e tests should find 0 testflight stonks at creation block 1`] = `[]`;

exports[`treasury-swap e2e tests should find 0 testflight stonks at next after creation block 1`] = `[]`;

exports[`treasury-swap e2e tests should find 1 testflight stonks at creation block 1`] = `
[
  Finding {
    "addresses": [],
    "alertId": "STONKS-ORDER-FULFILL",
    "description": "Stonks order [0x0D4d93c171452D8c58e7a8Cc2D70045eBf3D293c](https://etherscan.io/address/0x0D4d93c171452D8c58e7a8Cc2D70045eBf3D293c) was fulfill",
    "labels": [],
    "metadata": {
      "args": "?",
    },
    "name": "✅ Stonks: order fulfill",
    "protocol": "ethereum",
    "severity": 1,
    "source": {},
    "timestamp": 2023-12-31T00:00:00.000Z,
    "type": 4,
    "uniqueKey": "",
  },
]
`;

exports[`treasury-swap e2e tests should find 1 testflight stonks at next after creation block 1`] = `
[
  Finding {
    "addresses": [],
    "alertId": "STONKS-ORDER-FULFILL",
    "description": "Stonks order [0x0D4d93c171452D8c58e7a8Cc2D70045eBf3D293c](https://etherscan.io/address/0x0D4d93c171452D8c58e7a8Cc2D70045eBf3D293c) was fulfill",
    "labels": [],
    "metadata": {
      "args": "?",
    },
    "name": "✅ Stonks: order fulfill",
    "protocol": "ethereum",
    "severity": 1,
    "source": {},
    "timestamp": 2023-12-31T00:00:00.000Z,
    "type": 4,
    "uniqueKey": "",
  },
]
`;
